plugins/videobasedtracker/BeaconBasedPoseEstimator.cpp

```diff
@@ -212,10 +212,43 @@ namespace vbtracker {
         // do okay without using it, so leaving it out.
         // @todo Make number of iterations into a parameter.
         bool usePreviousGuess = false;
+        int maxIterations = 5;
+        cv::Mat inlierIndices;
         cv::solvePnPRansac(
             objectPoints, imagePoints, m_cameraMatrix, m_distCoeffs, m_rvec,
-            m_tvec, usePreviousGuess, 20, 8.0f,
-            static_cast<int>(objectPoints.size() - m_permittedOutliers));
+            m_tvec, usePreviousGuess, 5, 8.0f,
+            static_cast<int>(objectPoints.size() - m_permittedOutliers),
+            inlierIndices);
+
+        //==========================================================================
+        // Make sure we got all the inliers we needed.  Otherwise, reject this
+        // pose.
+        if (inlierIndices.rows < m_requiredInliers) {
+          return false;
+        }
+
+        //==========================================================================
+        // Reproject the inliers into the image and make sure they are actually
+        // close to the expected location; otherwise, we have a bad pose.
+        if (inlierIndices.rows > 0) {
+          std::vector<cv::Point3f>  inlierObjectPoints;
+          std::vector<cv::Point2f> inlierImagePoints;
+          for (int i = 0; i < inlierIndices.rows; i++) {
+            inlierObjectPoints.push_back(objectPoints[i]);
+            inlierImagePoints.push_back(imagePoints[i]);
+          }
+          std::vector<cv::Point2f> reprojectedPoints;
+          cv::projectPoints(inlierObjectPoints, m_rvec, m_tvec, m_cameraMatrix,
+            m_distCoeffs, reprojectedPoints);
+          for (size_t i = 0; i < reprojectedPoints.size(); i++) {
+            if (reprojectedPoints[i].x - inlierImagePoints[i].x > 4) {
```

Comment:
Magic number I'm tuning to figure out where to cut off reprojections.  Will give it a name.